workitem/workitem_repository.go

```diff
@@ -104,6 +104,10 @@ func (r *GormWorkItemRepository) Save(ctx context.Context, wi app.WorkItem) (*ap
 	if res.Version != wi.Version {
 		return nil, errors.NewVersionConflictError("version conflict")
 	}
+	// Changing the type of a work item is prohibited.
+	if res.Type != wi.Type {
+		return nil, errors.NewBadParameterError("type", wi.Type).Expected(res.Type)
```

Comment:
Generally, REST should silently ignore things it doesn't understand. In this case, we no longer 'undrestand' a PATCH of a basetype on an existing workitem. If we start failing on PATCH with a BaseType in the request we break all clients regardless of their intent or what they are updating(e.g. currently the UI will send the complete Object on a PATCH because they don't really understand PATCH).